src/routes/welcome/setup.tsx

```diff
@@ -9,100 +9,89 @@ import {
 import { Card, Spacer, useToasts } from "@arconnect/components";
 import type { JWKInterface } from "arweave/web/lib/wallet";
 import { jwkFromMnemonic } from "~wallets/generator";
-import { useLocation, useRoute } from "wouter";
 import { ArrowLeftIcon } from "@iconicicons/react";
 import browser from "webextension-polyfill";
 import * as bip39 from "bip39-web-crypto";
 import styled from "styled-components";
 import Arweave from "arweave";
-
-import GenerateDone from "./generate/done";
-import Confirm from "./generate/confirm";
-import Backup from "./generate/backup";
-
-import Password from "./load/password";
-import Wallets from "./load/wallets";
-import LoadDone from "./load/done";
-import Theme from "./load/theme";
 import { defaultGateway } from "~gateways/gateway";
 import Pagination, { Status } from "~components/Pagination";
 import { getWalletKeyLength } from "~wallets";
+import { useLocation } from "~wallets/router/router.utils";
+import type { CommonRouteProps } from "~wallets/router/router.types";
+
+// Shared:
+import { PasswordWelcomeView } from "./load/password";
+import { ThemeWelcomeView } from "./load/theme";
+
+// Generate:
+import { BackupWelcomeView } from "./generate/backup";
+import { ConfirmWelcomeView } from "./generate/confirm";
+import { GenerateDoneWelcomeView } from "./generate/done";
+
+// Load:
+import { WalletsWelcomeView } from "./load/wallets";
+import { LoadDoneWelcomeView } from "./load/done";
+import { Redirect } from "~wallets/router/components/redirect/Redirect";
+
+// Wallet generate pages:
+
+// TODO: Use a nested router instead:
```

Comment:
This could/should be implemented using `<Routes>`. See https://github.com/arconnectio/ArConnect/issues/567.